packages/plugin-truffle/src/utils/validations.ts

```diff
@@ -86,18 +87,41 @@ function checkForImportIdConsistency(
     for (const importDir of findAll('ImportDirective', ast)) {
       const importedUnitId = sourceUnitId[importDir.absolutePath];
       if (importedUnitId === undefined) {
-        // This can happen in two scenarios (I think):
+        // This can happen in three scenarios (we think):
         //
         // 1. There is more than one contract with the same name in different source files.
         //    Truffle only generates a single artifact.
         //    This scenario should have been detected before, and caused an error.
         //
-        // 2. A contract was imported from a dependency using artifacts.require.
+        // 2. An imported file has only other import statements without an actual contract/library/interface.
+        //    Truffle does not output the AST for this file.
+        //    If this looks like an OpenZeppelin interface that transitively imports an interface at a different path,
+        //    ignore this import.
+        //
+        // The code below corresponds to scenario 2.
+        let ignoreImport = false;
+        for (const sourceUnitPath of Object.keys(sourceUnitId) as Array<string>) {
```

Comment:
This doesn't look necessary:
```suggestion
        for (const sourceUnitPath of Object.keys(sourceUnitId)) {
```